url8d3c4d17f5_catseye_JaC64_tgz-pJ8-JaC64J8/src/com/dreamfabric/jac64/C64Screen.java

```diff
@@ -0,0 +1,2025 @@
+    }
+
+    monitor.info("IRQFlags: " + getIRQFlags());
+    monitor.info("NMIFlags: " + getNMIFlags());
+    monitor.info("CPU IRQLow: " + cpu.getIRQLow());
+    monitor.info("CPU NMILow: " + cpu.NMILow);
+    monitor.info("Current CPU cycles: " + cpu.cycles);
+    monitor.info("Next IO update: " + nextIOUpdate);
+  }
+
+  public void setSoundOn(boolean on) {
+   audioDriver.setSoundOn(on);
+  }
+
+  public void setStick(boolean one) {
+    keyboard.setStick(one);
+  }
+
+  public void registerHotKey(int key, int mod, String script, Object o) {
+    keyboard.registerHotKey(key, mod, script, o);
+  }
+
+  public void setKeyboardEmulation(boolean extended) {
+    monitor.info("Keyboard extended: " + extended);
+
+    keyboard.stickExits = !extended;
+    keyboard.extendedKeyboardEmulation = extended;
+  }
+
+  public void init(CPU cpu) {
+    super.init(cpu);
+
+    this.memory = cpu.getMemory();
+
+    c1541Chips = cpu.getDrive().chips;
+    c1541Chips.initIEC2(this);
+    c1541Chips = cpu.getDrive().chips;
+    c1541Chips.setObserver(this);
+
+
+    for (int i = 0, n = sprites.length; i < n; i++) {
+      sprites[i] = new Sprite();
+      sprites[i].spriteNo = i;
+    }
+
+    cia = new CIA[2];
+    cia[0] = new CIA(cpu, IO_OFFSET + 0xdc00, this);
+    cia[1] = new CIA(cpu, IO_OFFSET + 0xdd00, this);
+
+    tfe = new TFE_CS8900(IO_OFFSET + 0xde00);
+
+//  c1541 = new C1541(memory);
+//  c1541.addObserver(this);
+
+    keyboard = new Keyboard(this, cia[0], memory);
+    canvas = new C64Canvas(this, DOUBLE, keyboard);
+    canvas.addMouseMotionListener(this);
+    canvas.addMouseListener(this);
+
+    audioDriver = new AudioDriverSE();
+    audioDriver.init(44000, 22000);
+    setSID(RESID_6581);
+    charMemoryIndex = CPU.CHAR_ROM2;
+
+    for (int i = 0; i < SC_WIDTH * SC_HEIGHT; i++) {
+      mem[i] = cbmcolor[6];
+    }
+
+    mis = new MemoryImageSource(SC_WIDTH, SC_HEIGHT, mem, 0, SC_WIDTH);
+
+    mis.setAnimated(true);
+    mis.setFullBufferUpdates(true);
+    screen = canvas.createImage(mis);
+
+    //to fix bug http://developer.java.sun.com/developer/bugParade/bugs/4464723.html
+//  setRequestFocusEnabled(true);
+//  addMouseListener(new MouseAdapter() {
+//  public void mousePressed(MouseEvent e) {
+//  requestFocus();
+//  }
+//  });
+    initUpdate();
+  }
+
+  public void update(Object src, Object data) {
+    if (src != c1541Chips) {
+      // Print some kind of message...
+      message = (String) data;
+    } else {
+      updateDisk(src, data);
+    }
+  }
+
+  void restoreKey(boolean down) {
+    if (down) setNMI(KEYBOARD_NMI);
+    else clearNMI(KEYBOARD_NMI);
+  }
+
+  // Should be checked up!!!
+  private static final int[] IO_ADDRAND = new int[] {
+    0xd03f, 0xd03f, 0xd03f, 0xd03f,
+    0xd41f, 0xd41f, 0xd41f, 0xd41f,
+    0xd8ff, 0xd9ff, 0xdaff, 0xdbff, // Color ram
+    0xdc0f, 0xdd0f, 0xdeff, 0xdfff, // CIA + Expansion...
+  };
+
+  public int performRead(int address, long cycles) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 112 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.